lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -367,34 +373,37 @@ def _expand_root_partition(self, node: RemoteNode) -> None:
     def _delete_environment(self, environment: Environment, log: Logger) -> None:
         self._delete_nodes(environment, log)
 
-    def _delete_nodes(self, environment: Environment, log: Logger) -> None:
-        def _delete_node(node_ctx: NodeContext, wait_delete: bool) -> None:
-            hv = self._server.tools[HyperV]
-            vm_name = node_ctx.vm_name
-
-            # Reassign passthrough devices to host before VM is deleted
-            # This will be hot-unplug of device
-            if len(node_ctx.passthrough_devices) > 0:
-                self.device_pool.release_devices(node_ctx)
-
-            if wait_delete:
-                hv.delete_vm(vm_name)
-            else:
-                hv.delete_vm_async(vm_name)
-
-            assert node_ctx.serial_log_process
-            result = node_ctx.serial_log_process.wait_result()
-            log.debug(
-                f"{vm_name} serial log process exited with {result.exit_code}. "
-                f"stdout: {result.stdout}"
-            )
+    def _delete_node(
+        self, node_ctx: NodeContext, wait_delete: bool, log: Logger
+    ) -> None:
+        hv = self._server.tools[HyperV]
+        vm_name = node_ctx.vm_name
+
+        # Reassign passthrough devices to host before VM is deleted
+        # This will be hot-unplug of device
+        if len(node_ctx.passthrough_devices) > 0:
+            self.device_pool.release_devices(node_ctx)
 
+        if wait_delete:
+            hv.delete_vm(vm_name)
+        else:
+            hv.delete_vm_async(vm_name)
+
+        assert node_ctx.serial_log_process
+        result = node_ctx.serial_log_process.wait_result()
+        log.debug(
+            f"{vm_name} serial log process exited with {result.exit_code}. "
+            f"stdout: {result.stdout}"
+        )
+
+    def _delete_nodes(self, environment: Environment, log: Logger) -> None:
         run_in_parallel(
             [
                 partial(
-                    _delete_node,
+                    self._delete_node,
```

Comment:
Did you test this? Now that you made it a member function, I believe it requires one additional argument: `self`.

Reply:
I thought the same too. To my surprise, this works only without `self` parameter. It throws error saying too many arguments when `self` is additionally passed. This could be because we are passing the function already as `self.<method>`

Reply:
This change should work.